opsmngr/all_clusters.go

```diff
@@ -0,0 +1,65 @@
+package opsmngr
+
+import (
+	"context"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	allClustersBasePath = "clusters"
+)
+
+// AllClustersService is an interface for interfacing with Clusters in MongoDB Ops Manager APIs
+type AllClustersService interface {
+	List(ctx context.Context) ([]*AllClustersProject, *atlas.Response, error)
+}
+
+type AllClustersServiceOp struct {
+	Client atlas.RequestDoer
+}
+
+type AllClustersProject struct {
+	GroupName string               `json:"groupName,omitempty"`
+	OrgName   string               `json:"orgName,omitempty"`
+	PlanType  string               `json:"planType,omitempty"`
+	GroupID   string               `json:"groupId,omitempty"`
+	OrgID     string               `json:"orgId,omitempty"`
+	Tags      []string             `json:"tags,omitempty"`
+	Clusters  []AllClustersCluster `json:"clusters,omitempty"`
```

Comment:
[q] do you expect any of them to be empty? also consider that and empty array is not empty by go definitions of serialization